crates/bevy_asset/src/loader_builders.rs

```diff
@@ -74,47 +209,105 @@ impl<'ctx, 'builder> NestedLoader<'ctx, 'builder> {
         self.with_transform(move |meta| meta_transform_settings(meta, &settings))
     }
 
-    /// Specify the output asset type.
+    // convert between `T`s
+
+    /// When [`load`]ing, you must pass in the asset type as a type parameter
+    /// statically.
+    ///
+    /// If you don't know the type statically (at compile time), consider
+    /// [`with_dynamic_type`] or [`with_unknown_type`].
+    ///
+    /// [`load`]: Self::load
+    /// [`with_dynamic_type`]: Self::with_dynamic_type
+    /// [`with_unknown_type`]: Self::with_unknown_type
     #[must_use]
-    pub fn with_asset_type<A: Asset>(mut self) -> Self {
-        self.asset_type_id = Some(TypeId::of::<A>());
-        self
+    pub fn with_static_type(self) -> NestedLoader<'ctx, 'builder, StaticTyped, M> {
+        NestedLoader {
+            load_context: self.load_context,
+            meta_transform: self.meta_transform,
+            typing: StaticTyped { _priv: () },
+            mode: self.mode,
+        }
     }
 
-    /// Specify the output asset type.
+    /// When [`load`]ing, the loader will attempt to load an asset with the
+    /// given [`TypeId`].
+    ///
+    /// [`load`]: Self::load
     #[must_use]
-    pub fn with_asset_type_id(mut self, asset_type_id: TypeId) -> Self {
-        self.asset_type_id = Some(asset_type_id);
-        self
+    pub fn with_dynamic_type(
+        self,
+        asset_type_id: TypeId,
+    ) -> NestedLoader<'ctx, 'builder, DynamicTyped, M> {
+        NestedLoader {
+            load_context: self.load_context,
+            meta_transform: self.meta_transform,
+            typing: DynamicTyped { asset_type_id },
+            mode: self.mode,
+        }
     }
 
-    /// Load assets directly, rather than creating handles.
+    /// When [`load`]ing, we will infer what type of asset to load from
+    /// metadata.
+    ///
+    /// [`load`]: Self::load
     #[must_use]
-    pub fn direct<'c>(self) -> DirectNestedLoader<'ctx, 'builder, 'c> {
-        DirectNestedLoader {
-            base: self,
-            reader: None,
+    pub fn with_unknown_type(self) -> NestedLoader<'ctx, 'builder, UnknownTyped, M> {
+        NestedLoader {
+            load_context: self.load_context,
+            meta_transform: self.meta_transform,
+            typing: UnknownTyped { _priv: () },
+            mode: self.mode,
+        }
+    }
+
+    // convert between `M`s
+
+    /// When [`load`]ing, create only asset handles, rather than loading the
```

Comment:
Perhaps "rather than *returning* the actual asset". It sounds a little strange, like we don't load the asset at all with deferred.